### PR TITLE
✨(frontend) add page 404 not found

### PR DESCRIPTION
**Pull request by rlecellier**
We're currently using react router default 404 page. Let's design our own.

image

### CODE REVIEW

File: src/frontend/js/pages/DashboardPageNotFound/_styles.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  &__title {
+    color: r-theme-val(body-content, text-color);
+    font-size: rem-calc(40px);
+  }
+
+  &__subtitle {
+    color: r-theme-val(body-content, text-color);
+    font-size: rem-calc(30px);
+  }
+}
```
Comment (jbpenrath) [inline src/frontend/js/pages/DashboardPageNotFound/_styles.scss:23]: A blank line is missing at EOF

File: src/frontend/js/pages/DashboardPageNotFound/index.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        </p>
+        <p className="page-not-found__link">
+          <Link to={redirectPath}>
+            <FormattedMessage {...messages.pageNotFoundHomeLink} />
+          </Link>
+        </p>
+      </div>
+    </div>
+  );
+};
```
Comment (jbpenrath) [inline src/frontend/js/pages/DashboardPageNotFound/index.tsx:54]: IMO this is overengineer. I'll just put a link to the home page like the link label refers.

**Review by jbpenrath — COMMENTED** [review]
- Add an entry in the CHANGELOG
- Add tests

File: src/frontend/js/widgets/Dashboard/index.spec.tsx [hunk 3]
```diff
@@ -196,4 +196,12 @@ describe('<Dashboard />', () => {
     const sidebar = screen.getByTestId('dashboard__sidebar');
     getByText(sidebar, user.full_name!, { exact: false });
   });
+
+  it("should redirect to 404 page when route doesn't exists", async () => {
```
Comment (jbpenrath) [inline src/frontend/js/widgets/Dashboard/index.spec.tsx:200]:
```suggestion
  it("should redirect to 404 page when route doesn't exist", async () => {
```

File: src/frontend/js/widgets/Dashboard/utils/dashboardRoutes.tsx [hunk 4]
```diff
@@ -24,10 +25,21 @@ export interface DashboardRouteHandle {
...
       path: '/',
       element: <DashboardLayoutRoute />,
       children: [
+        {
+          path: '/page-not-found',
+          element: <DashboardPageNotFound />,
+          handle: {
+            renderLayout: true,
+          },
+        },
```
[the remark this answers is not in the document]
Comment (rlecellier) [reply src/frontend/js/widgets/Dashboard/utils/dashboardRoutes.tsx]: I keep the page not found route under `DashboardLayoutRoute` to have the back button and centered content .

File: src/frontend/js/widgets/Dashboard/components/DashboardLayout/index.tsx [hunk 5]
```diff
@@ -19,7 +19,9 @@ export const DashboardLayout = ({
   const location = useLocation();
   return (
     <div className={c('dashboard', className)}>
-      <div className="dashboard__sidebar">{sidebar || <LearnerDashboardSidebar />}</div>
+      <div className="dashboard__sidebar">
+        {sidebar !== undefined ? sidebar : <LearnerDashboardSidebar />}
+      </div>
```
Comment (jbpenrath) [inline src/frontend/js/widgets/Dashboard/components/DashboardLayout/index.tsx:24]: IMO this is hard to understand, WDYT to explicitly use a prop `hideSidebar` ?

**Review by jbpenrath — APPROVED** [review]

**Pull request merged**